Fixed testsuite to work with ISPN 10 Server.

andyuk1986 opened the pull request:
Fixed the tests and configuration for running ispn server according to ISPN 10.
Changed the calls to Buffer according to node 12 version recommendation.

pruivo commented:
I have 14 failures, all related to TLS
`Error: [Client network socket disconnected before secure TLS connection was established] is not any of: [SELF_SIGNED_CERT_IN_CHAIN,self signed certificate]`

andyuk1986 commented:
@pruivo @karesti exactly - I have fixed all tests besides the SSL ones. I have fixed the config for SSL 3 tests are passing, but still 14 are failing and  I lost there completely. We have agreed with Katia that I will send a PR and she will look into SSL tests, i.e. spec/infinispan_ssl_spec.js

pruivo commented on spec/utils/testing.js:
```diff
@@ -16,6 +16,7 @@ var ispn = require('../../lib/infinispan');
 var u = require('../../lib/utils');
 var protocols = require('../../lib/protocols');
 
+exports.serverDirName = "infinispan-js-serverng";
```
this is not always true. when `run-servers.sh --ci` is run, the server dir is another location.

```
Error: spawn /tmp/infinispan-js-serverng//bin/server.sh ENOENT
    at Process.ChildProcess._handle.onexit (internal/child_process.js:232:19)
    at onErrorNT (internal/child_process.js:407:16)
    at process._tickCallback (internal/process/next_tick.js:63:19)
```

pruivo commented on run-servers.sh:
```diff
@@ -0,0 +1,148 @@
+             echo "Admin user added."
+        fi
+
+        cp -r ${SERVER_HOME}/server ${SERVER_TMP}/${dirName}
```
this copy needs to be done with or without the `--ci` option. 
Otherwise, if you run `run-servers.sh --ci` is will fail to execute.

pruivo commented on run-servers.sh:
```diff
@@ -0,0 +1,148 @@
+
+
+    if [[ ${isCi} = "--ci" ]]; then
+      nohup $SERVER_TMP/bin/server.sh -c ${confPath} "${portStr:-""}" --node-name=${nodeName} ${jvmParam:-} &
```
`-s` options missing and remove `"` between `${portStr:-""}`
`nohup $SERVER_TMP/bin/server.sh -c ${confPath} -s ${SERVER_TMP}/${nodeName} ${portStr:-""} --node-name=${nodeName} ${jvmParam:-} &`

andyuk1986 commented:
@pruivo @karesti thanks for the review. I have changed the run-server.sh accordingly so that it works with both ci and standalone. Also I have made changes to Jenkinsfile so that the trigger works properly. And I have also made some changes to README.md (changed the standalone.sh to server.sh with proper parameters, as well as I have went through all links and fixed the broken ones).
